src/plugins/kibana/public/settings/sections/indices/add_data_steps/pipeline_setup/lib/pipeline.js

```diff
@@ -108,24 +107,44 @@ export default class Pipeline {
     return result;
   }
 
-  // Updates the state of the pipeline and processors with the results
-  // from an ingest simulate call.
-  applySimulateResults(results) {
-    //update the outputObject of each processor
-    results.forEach((result) => {
+  _updateProcessorOutputs(simulateResults) {
+    simulateResults.forEach((result) => {
       const processor = this.getProcessorById(result.processorId);
 
       processor.outputObject = _.get(result, 'output');
       processor.error = _.get(result, 'error');
     });
+  }
 
-    //update the inputObject of each processor
-    results.forEach((result) => {
-      const processor = this.getProcessorById(result.processorId);
+  //Updates the error state of the pipeline and its processors
+  //If a pipeline compile error is returned, lock all processors but the error
+  //If a pipeline data error is returned, lock all processors after the error
+  _updateErrorState() {
+    this.hasCompileError = _.some(this.processors, (processor) => {
+      return _.get(processor, 'error.compile');
+    });
+    _.forEach(this.processors, processor => {
+      processor.locked = false;
+    });
 
+    const errorIndex = _.findIndex(this.processors, 'error');
+    if (errorIndex === -1) return;
+
+    _.forEach(this.processors, (processor, index) => {
+      if (this.hasCompileError && index !== errorIndex) {
+        processor.locked = true;
```

Comment:
@bargs, There is only one processor that is in error at any given time. This is represented by the errorIndex variable in the above code. ```this.hasCompileError```, which is a property of the whole pipeline, stores whether the processor that is in error (if one exists) is in error because of a compile error. The above conditional sets all processors except the one in error to locked. 

Please keep trying to figure out how you were able to get into your mentioned state, because by my described logic, this should not be possible.